assets/js/push.js

```diff
@@ -122,6 +148,45 @@ jQuery( window ).on( 'load', () => {
 	function distributorMenuEntered() {
 		distributorMenuItem.focus();
 		document.body.classList.toggle( 'distributor-show' );
+
+		if ( distributorPushWrapper.classList.contains( 'loaded' ) ) {
+			return;
+		}
+
+		distributorPushWrapper.classList.remove( 'message-error' );
+		distributorPushWrapper.classList.add( 'loaded' );
+
+		const data = {
+			action: 'dt_load_connections',
+			loadConnectionsNonce: dt.loadConnectionsNonce,
+			postId: dt.postId
+		};
+
+		const xhr = dt.usexhr ? { withCredentials: true } : false;
+
+		jQuery.ajax( {
+			url: dt.ajaxurl,
+			xhrFields: xhr,
+			method: 'post',
+			data: data
+		} ).done( ( response ) => {
+			if ( ! response.success || ! response.data ) {
+				distributorPushWrapper.classList.remove( 'loaded' );
+				distributorPushWrapper.classList.add( 'message-error' );
+				return;
+			}
+
+			dtConnections = response.data;
+
+			distributorPushWrapper.innerHTML = processTemplate( 'dt-show-connections' )( {
```

Comment:
Let's just leave a comment for now and revisit the overall approach later. For instance, I think it might be nice to store the last 3-5 sites that were distributed to and front-load those.

```suggestion
			// Allowing innerHTML because processTemplate escapes values
			distributorPushWrapper.innerHTML = processTemplate( 'dt-show-connections' )( {
```